src/Generator/Normalizer/NormalizerGenerator.php

```diff
@@ -72,20 +72,27 @@ protected function createSupportsNormalizationMethod($modelFqdn)
     protected function createNormalizeMethod($modelFqdn, Context $context, $properties)
     {
         $dataVariable = new Expr\Variable('data');
-        $statements     = [
-            new Expr\Assign($dataVariable, new Expr\New_(new Name("\\stdClass"))),
+        $statements = [
+            new Expr\Assign($dataVariable, new Expr\New_(new Name('\\stdClass'))),
         ];
 
+        /** @var Property $property */
         foreach ($properties as $property) {
-            $propertyVar                                 = new Expr\MethodCall(new Expr\Variable('object'), $this->getNaming()->getPrefixedMethodName('get', $property->getName()));
+            $propertyVar = new Expr\MethodCall(new Expr\Variable('object'), $this->getNaming()->getPrefixedMethodName('get', $property->getName()));
             list($normalizationStatements, $outputVar) = $property->getType()->createNormalizationStatement($context, $propertyVar);
 
+            $normalizationStatements[] = new Expr\Assign(new Expr\PropertyFetch($dataVariable, sprintf("{'%s'}", $property->getName())), $outputVar);
+
+            if ($property->isNullable()) {
```

Comment:
This way the data goes through all kinds of checks unnecessarily IMO. If we preserve the if statement and add an else then null can be set directly and all other checks (string, integer, etc) can be skipped.